### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by qiarie**

### CODE REVIEW

File: opensrp-child/src/test/java/org/smartregister/child/util/ChildJsonFormUtilsTest.java [hunk 1]
```diff
@@ -691,4 +691,68 @@ public void testUpdateClientAttribute() throws Exception {
...
+        Assert.assertEquals("Birth Registration", client.getEvent().getEventType());
+    }
+
+    @Test
+    public void testFieldsReturnsJSONArrayForNonEmptyFormString() {
+        JSONObject form = JsonFormUtils.toJSONObject(registrationForm);
+        String step = "step1";
+
+        JSONArray formFields = ChildJsonFormUtils.fields(form, step);
+        Assert.assertNotNull(formFields);
```
Comment (ekigamba) [inline opensrp-child/src/test/java/org/smartregister/child/util/ChildJsonFormUtilsTest.java:725]: I think we can optionally assert the number of fields returned here (JSONArray size). What do you think?

File: opensrp-child/src/test/java/org/smartregister/child/util/ChildJsonFormUtilsTest.java [hunk 2]
```diff
@@ -691,4 +691,68 @@ public void testUpdateClientAttribute() throws Exception {
...
+
+        ChildEventClient client = ChildJsonFormUtils.processChildDetailsForm(jsonFormString, formTag);
+        Assert.assertNotNull(client);
+        Assert.assertNotNull(client.getClient());
+        Assert.assertNotNull(client.getEvent());
+        Assert.assertEquals("Birth Registration", client.getEvent().getEventType());
```
[the remark this answers is not in the document]
Comment (qiarie) [reply opensrp-child/src/test/java/org/smartregister/child/util/ChildJsonFormUtilsTest.java:716]: This is in order. Will try and capture as much as possible

File: opensrp-child/src/test/java/org/smartregister/child/util/ChildJsonFormUtilsTest.java (hunk 1 above)
Comment (qiarie) [reply opensrp-child/src/test/java/org/smartregister/child/util/ChildJsonFormUtilsTest.java:725]: Noted for action

File: opensrp-child/src/test/java/org/smartregister/child/util/ChildJsonFormUtilsTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (qiarie) [reply opensrp-child/src/test/java/org/smartregister/child/util/ChildJsonFormUtilsTest.java:767]: Added a few more checks

**Review by ekigamba — APPROVED** [review]

**Pull request merged**